src/runtime/recipe/recipe-resolver.ts

```diff
@@ -76,9 +84,16 @@ export class ResolveWalker extends RecipeWalker {
         default:
           throw new Error(`unexpected fate ${handle.fate}`);
       }
-      mappable = storeById ? [storeById] : [];
+      if (storeById) {
+        mappable = [storeById];
+      } else {
+        return error('cannot find associated store');
+      }
     }
 
+    if (mappable.length === 0) {
+      return error('No mappable connections');
```

Comment:
This doesn't seem like a message one would understand outside of the scope of this method.

Maybe something like: "Cannot find a handle matching requested type and tags."

Reply:
Thanks, that's much clearer.